src/fsharp/PostInferenceChecks.fs

```diff
@@ -959,11 +959,10 @@ and CheckAttribs cenv env (attribs: Attribs) =
     // Check for violations of allowMultiple = false
     let duplicates = 
         tcrefs 
-        |> Seq.groupBy (fun (tcref,_) -> tcref.Stamp) 
-        |> Seq.map (fun (_,elems) -> List.last (List.ofSeq elems), Seq.length elems) 
-        |> Seq.filter (fun (_,count) -> count > 1) 
-        |> Seq.map fst 
-        |> Seq.toList
+        |> List.groupBy (fun (tcref,_) -> tcref.Stamp) 
+        |> List.map (fun (_,elems) -> List.last elems, List.length elems) 
+        |> List.filter (fun (_,count) -> count > 1) 
+        |> List.map fst 
```

Comment:
I'd rollback this change. Long `List.xxx` pipes result with N * (list length) allocations, `Seq.xxx` do not.